### ISSUE TITLE
Meet accessibility standards checked by common benchmarks

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#52532]
### Describe the Issue

A software engineer and freeCodeCamp donor, Jonathan Pool, has created a suite of accessibility tests he runs against the landing pages of charities.

Currently freeCodeCamp is the 4th most accessible charity landing page on the web, but I see some ways we can make it to the number one spot.

Here's his annual report: https://jpdev.pro/jpdev/blog/entries/0023/index.html

Here is the code this donor runs to create his report: https://github.com/jrpool/testaro

Once we've finished our curriculum initiatives for 2023, I propose we embark on driving down this score for our landing page, and running this suite against other pages of freeCodeCamp as well to identify further places we can improve.

### Affected Page

https://www.freecodecamp.org

### Steps to Reproduce

Upon running the tests in the GitHub article, you'll get results that will look like this output JSON: https://jpdev.pro/jpdev/blog/entries/0023/reports/231122T235231-ts37a-target.json

Those are the accessibility issues we'll want to improve upon.

### Expected behavior

We want to get as low a score as possible

### System

Desktop

**Comment by bbsmooth** [comment]
Hey, fourth best is pretty darn good. I got to meet Jonathon Pool earlier this year and he's a good guy and well known in the accessibility community. 

I'll be happy to look into this report when I get a chance and see what can be improved. Also, keep in mind, that while I'm sure this tool is very good, it doesn't mean that there aren't possible errors in it. For example:

- The second highest issue we got docked for is "page title missing". I just checked, and there is definitely a `title` element in the `head` of our landing page.
- The third highest issue is "page language missing". Again, I just checked and we definitely have a `lang="en"` on the opening `html` tag of our landing page.
- The fourth highest issue is "h1 missing" and there is definitely an `h1` element on our landing page.

Just those three false positives drop our score `341` points which puts us in third place and gets us `21` points away from second, which I'm guessing we can easily make up with several other false positives I see further down the list. (For example: "headings missing", "main missing", "no landmarks".) Of course I'm sure there are false positives for the other sites too, so perhaps we will never jump to number one.

Don't get me wrong, automated tools definitely have their role in accessibility testing. But using them as the sole basis for determining accessibility is not recommended, and I'm sure Jonathon does not recommend it either. I'll admit, this is sort of fun and rewarding to see a comparison like this and see us near the top of the list. But based on the false positives I am seeing in the report for fCC I am slightly disappointed in the quality of the testing. Perhaps there is some technical reason why this tool doesn't think we have all the false positives I outlined above? I think we would want to figure that out first before we decided to rely on it to test other pages.

**Comment by a2937** [comment]
I feel like some of these tests are invalid. Is there any way to actually see the source code of the tests that FreeCodeCamp was ran against?

**Comment by a2937** [comment]
Okay I found something interested. The testing seems to care about line height being 1.5rem. That could be an easy fix.

**Comment by bbsmooth** [comment]
> The testing seems to care about line height being 1.5rem

Yes, it is best practice to use unitless numbers to set the line-height, as it could potentially create an accessibility issue (see [MDN's example](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height#prefer_unitless_numbers_for_line-height_values)). So ya, fixing that will give a small handful of points back for this particular tool. Best practices are there for a reason, right?

Probably would be worthwhile to run our CSS through a validator and fix all of those errors as well. As I mentioned in my previous comment, I doubt many of the CSS errors are affecting our accessibility. It's just probably a good thing to do.

**Comment by a2937** [comment]
Do you think the [stylelint](https://www.npmjs.com/package/stylelint) package could be a good fit for FreeCodeCamp?

**Comment by a2937** [comment]
It seemed pretty opinionated when I tried experimenting with it. But that's probably a really good thing.  
![Preview](https://github.com/freeCodeCamp/freeCodeCamp/assets/9373245/f9aa018e-ed19-4416-babd-294a5dff86e0)

**Comment by bbsmooth** [comment]
@ahmaxed You are correct, they are not sufficient. Automated tools are good for finding issues that are binary enough to be either true or false. There are many accessibility issues that meet those requirements, such as color contrast, or checking if inputs have accessible names. But realistically, only perhaps a third of the possible accessibility issues will be covered by these tools. Not terrible, but not sufficient. And as I pointed out back in December when this issue was originally opened, some of these tools are either too aggressive and produce false positives, or are just wrong.

I definitely encourage the use of automated tools to assist with accessibility testing. They can help you quickly find the "low hanging fruit" that needs to be fixed. By all means, run them against the site as much as you like and fix the legitimate issues they discover. I only ask that we just be sure to keep reminding ourselves that those perfect accessibility scores they give us are really an illusion and should not be used as an indication of how accessible the site really is.

**Comment by Sembauke** [comment]
I am going to close this as too broad. for contributors: If any accessibility issues come to mind please open individual issues that are small and easy to manage.
